utils/k8s/deprovision_deployment.yml

```diff
@@ -10,37 +9,57 @@
 - block:
 
     - name: Check if the application to be deleted is running.
-      shell: kubectl get pods -n {{ app_ns }} -l {{ app_label }} --no-headers -o custom-columns=:status.phase
-      args:
-        executable: /bin/bash
+      k8s_facts:
+        kind: Pod
+        label_selectors:
+          - "{{ app_label }}"
+        namespace: "{{ app_ns }}"
       register: result
-      until: "'Running' in result.stdout"
+      until: "result.resources.0.status.phase == 'Running'"
```

Comment:
The previous check would ensure all pods belonging having the specific label in that ns is in running state. Will the replacement work the same way (prompted by the `resources.0.status` reference above)